app.py

```diff
@@ -69,8 +69,45 @@ def daily():
 
 	graph_data = OrderedDict() # The data used in the render template (see format below)
 
-	if tab == "Location":
-		return render_template('daily_location.html', **locals())  # Location tab uses separate template to display map
+	if tab == "Location":  # Location tab uses separate template to display map
+		try:
+			# the times, in seconds from 0 on the day, for which to display data
+			starttime = int(request.args.get('starttime', default=''))
+			endtime = int(request.args.get('endtime', default=''))
+		except ValueError:
+			# default to whole day if not provided by GET parameter
+			starttime = 0
+			endtime = 86400
+
+		# get list of latitudes; lat_gen is a generator of document snapshots, but will only yield one snapshot
+		# with how the database is currently set up
+		lat_gen = db.collection(DATABASE_COLLECTION).document(date_str).collection('gps_lat').stream()
+
+		# avoid throwing an error if there's no data for the day (lat_gen will yield no values)
+		try:
+			lat_reading_dict = next(lat_gen).to_dict()["seconds"]  # reading_dict format: {'second': reading}, ex. {'10': 334}
+		except StopIteration:
+			location_pairs = None
+			return render_template('daily_location.html', **locals())
+
+		lat_reading_list = sorted({int(k): v for k, v in lat_reading_dict.items() if starttime <= int(k) <= endtime}.items())
+		sec_list, lat_list = zip(*lat_reading_list)
+
+		# get list of longitudes
+		lon_gen = db.collection(DATABASE_COLLECTION).document(date_str).collection('gps_lon').stream()
```

Comment:
Perhaps it would be worth writing a function to do this process for latitude and longitude instead of duplicating the code